### DISCUSSION TITLE
Filter in realtime update for ids in array

### DISCUSSION THREAD

**Discussion by rmrbytes** [supabase/supabase#11552] [Questions] [answered]
I have set up realtime update to a table. I would like to apply a filter condition so that only records that are part of some array needs to trigger the update

My channel subscription is like this:

```
supabase
      .channel('changes')
      .on(
        'postgres_changes',
        {
          event: 'UPDATE',
          schema: 'public',
          table: 'table_name',
          filter: ' ' // here would like to check for ids part of an array
        },
        ((payload) => { 
            // do something
        })
      )
      .subscribe()

```

I tried to look into the documentation but could not find any examples. Thanks

**Comment by GaryAustin1** [comment]
No.  You can only use a few filters right now.   .eq(), .lt(), .gt(), .gte(), .lte() and .neq().

You would need to do a separate subscription for each id in your array, or use RLS (if it applies) to do what you want.

**Reply by rmrbytes** [reply]
Thanks @GaryAustin1 . I am applying RLS. However in my use-case the realtime updates are for those records that the user has "subscribed" to (its a pub-sub app). Hence it is relevant. It is likely that the user has subscribed to a very small number of data that they have access to (via RLS).

I did try `filter:'id=ANY(${ids_arr})'` but it did not work

**Reply by GaryAustin1** [reply]
Just the single value filters I listed are available.

**Comment by rmrbytes** [comment]
Hopefully, it will get supported in the future ... meanwhile, I am applying the update logic to the app by checking the payload and comparing it to the array

**Comment by w3b6x9** [comment, marked as answer]
@rrrepos you can follow the development here: https://github.com/supabase/realtime/issues/401

**Comment by rmrbytes** [comment]
Thanks Wen. Will follow it

**Comment by w3b6x9** [comment]
Update: we have merged this and waiting to release: https://github.com/supabase/realtime/pull/467
